controllers/multiclusterhub_controller.go

```diff
@@ -1117,55 +1117,7 @@ func (r *MultiClusterHubReconciler) installCRDs(reqLogger logr.Logger, m *operat
 		return CRDRenderReason, err
 	}
 
-	files, err := os.ReadDir(crdDir)
-	if err != nil {
-		err := fmt.Errorf("unable to read CRD files from %s : %s", crdDir, err)
-		reqLogger.Error(err, err.Error())
-		return CRDRenderReason, err
-	}
-
-	crds := make([]*unstructured.Unstructured, 0, len(files))
-	errs := make([]error, 0, len(files))
-	for _, file := range files {
-		fileName := file.Name()
-		if filepath.Ext(fileName) != ".yaml" {
-			continue
-		}
-
-		path := path.Join(crdDir, fileName)
-		src, err := ioutil.ReadFile(filepath.Clean(path)) // #nosec G304 (filepath cleaned)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("error reading file %s : %s", fileName, err))
-			continue
-		}
-
-		crd := &unstructured.Unstructured{}
-		if err = yaml.Unmarshal(src, crd); err != nil {
-			errs = append(errs, fmt.Errorf("error unmarshalling file %s to unstructured: %s", fileName, err))
-			continue
-		}
-
-		// Check that it is actually a CRD
-		crdKind, _, err := unstructured.NestedString(crd.Object, "spec", "names", "kind")
-		if err != nil {
-			errs = append(errs, fmt.Errorf("error getting Kind field from %s: %s", fileName, err))
-			continue
-		}
-		crdGroup, _, err := unstructured.NestedString(crd.Object, "spec", "group")
-		if err != nil {
-			errs = append(errs, fmt.Errorf("error getting Group field for %s : %s", fileName, err))
-			continue
-		}
-
-		if crd.GetKind() != "CustomResourceDefinition" || crdKind == "" || crdGroup == "" {
-			errs = append(errs, fmt.Errorf("error verifying file %s is a crd", fileName))
-			continue
-		}
-
-		utils.AddInstallerLabel(crd, m.GetName(), m.GetNamespace())
-		crds = append(crds, crd)
-	}
-
+	crds, errs := renderer.RenderCRDs(crdDir)
```

Comment:
This has us now installing CRDs when the MCH is applied correct? Have we decided that installing them when the operator first runs is not what we want to do?

Reply:
Yeah I left it in the control loop just to have it in one place and because that's what we were originally doing. In MCE we don't really have a choice because we define watches on CRDs we apply so we have to apply them before the controller starts. In MCH we haven't had that problem.